packages/graphql/test/schema.test.ts

```diff
@@ -0,0 +1,37 @@
+import type { Namespace } from "@typespec/compiler";
+import { expectDiagnosticEmpty } from "@typespec/compiler/testing";
+import { describe, expect, it } from "vitest";
+import { getSchema } from "../src/lib/schema.js";
+import { compileAndDiagnose } from "./test-host.js";
+
+describe("@schema", () => {
+  it("Creates a schema with no name", async () => {
+    const [program, { TestNamespace }, diagnostics] = await compileAndDiagnose<{
+      TestNamespace: Namespace;
+    }>(`
+      @schema
+      @test namespace TestNamespace {}
+    `);
+    expectDiagnosticEmpty(diagnostics);
+
+    const schema = getSchema(program, TestNamespace);
+
+    expect(schema?.type).toBe(TestNamespace);
+    expect(schema?.name).toBeUndefined();
+  });
+
+  it("Creates a schema with a specified name", async () => {
+    const [program, { TestNamespace }, diagnostics] = await compileAndDiagnose<{
+      TestNamespace: Namespace;
+    }>(`
+      @schema({name: "MySchema"})
```

Comment:
could we add a test when an invalid property is set?

```
@schema({foo: "bar"})
```

Reply:
Because of https://github.com/pinterest/typespec/pull/1/files/c0fff8ab630090bc0902ec48f797e0ac87683317#r1849148838, we no longer need to test this — it's inbuilt TypeSpec compiler behavior.